docs/source/task_guides/token-classification-lora.mdx

```diff
@@ -0,0 +1,372 @@
+
+Initialize a tokenizer and make sure you set `is_split_into_words=True` because the text sequence has already been split into words. However, this doesn't mean it is tokenized yet (even though it may look like it!), and you'll need to further tokenize the words into subwords.
+
+You'll also need to write a function to:
+
+1. Map each token to their respective word with the [`~transformers.BatchEncoding.word_ids`] method.
+2. Ignore the special tokens by setting them to `-100`.
+3. Label the first token of a given entity.
+
+```py
+def tokenize_and_align_labels(examples):
+    tokenized_inputs = tokenizer(examples["tokens"], truncation=True, is_split_into_words=True)
+
+    labels = []
+    for i, label in enumerate(examples[f"tags"]):
+        word_ids = tokenized_inputs.word_ids(batch_index=i)
+        previous_word_idx = None
+        label_ids = []
+        for word_idx in word_ids:
+            if word_idx is None:
+                label_ids.append(-100)
+            elif word_idx != previous_word_idx:
+                label_ids.append(label[word_idx])
+            else:
+                label_ids.append(-100)
+            previous_word_idx = word_idx
+        labels.append(label_ids)
+
+    tokenized_inputs["labels"] = labels
+    return tokenized_inputs
+```
+
+Use [`~datasets.Dataset.map`] to apply the `tokenize_and_align_labels` function to the dataset:
+
+```py
+tokenized_bionlp = bionlp.map(tokenize_and_align_labels, batched=True)
+```
+
+Finally, create a data collator to pad the examples to longest length in a batch:
+
+```py
+data_collator = DataCollatorForTokenClassification(tokenizer=tokenizer)
+```
+
+## Train
+
+Now you can create a [`PeftModel`]. Start by loading the base `bloom-560m` model, the number of expected labels, and the `id2label` and `label2id` dictionaries:
+
+```py
+id2label = {
+    0: "O",
+    1: "B-DNA",
+    2: "I-DNA",
+    3: "B-protein",
+    4: "I-protein",
+    5: "B-cell_type",
+    6: "I-cell_type",
+    7: "B-cell_line",
+    8: "I-cell_line",
+    9: "B-RNA",
+    10: "I-RNA",
+}
+label2id = {
+    "O": 0,
+    "B-DNA": 1,
+    "I-DNA": 2,
+    "B-protein": 3,
+    "I-protein": 4,
+    "B-cell_type": 5,
+    "I-cell_type": 6,
+    "B-cell_line": 7,
+    "I-cell_line": 8,
+    "B-RNA": 9,
+    "I-RNA": 10,
+}
+
+model = AutoModelForTokenClassification.from_pretrained(
+    "roberta-large", num_labels=11, id2label=id2label, label2id=label2id
+)
+```
+
+Define the [`LoraConfig`] with:
+
+- `task_type`, token classification
+- `r`, the dimension of the low-rank matrices
+- `lora_alpha`, scaling factor for the weight matrices
+- `lora_dropout`, dropout probability of the LoRA layers
+- `bias`, set to `all` to train all bias parameters
+
+<Tip>
+
+💡 The weight matrix is scaled by `lora_alpha/r` so changing one or the other will affect the number of trainable parameters. In the paper, the authors set `lora_alpha` equal to `r` to reduce the need to tune the hyperparameters when changing the value of `r`.
```

Comment:
Actually, the delta activations from LoRA are scaled as per alpha/r. So, higher alpha means that higher weightage is given to the LoRA activations and as such the gradients are also scaled accordingly. MAM paper uses LoRA and Scaled Parallel Adapter with alpha being 4X compared to r. The LoRA paper itself uses alpha ranging from 1X-8X that of r. 

Alpha doesn't change the number of trainable parameters, rather functions as a scaling factor.

Reply:
Recommended way would be to first see the performance without setting bias, so that pretrained model is completely frozen to keep the generalization of it intact. If te performance doesn't meet the requirements, then they can experiment with setting bias to `lora_only` and as a final step to `all`

Reply:
Really interesting, thanks for this!